### PR TITLE
implement createCircle mutation in FE

### PR DESCRIPTION
**Pull request by CryptoGraffe**
Fixes: #629 

This also:
* Makes a number of columns that seemed non-nullable actually non-nullable to match the typescript interfaces in frontend
* removes HASURA_ENABLED flag in FE since we want to default to using it and delete the old code
* removes concept or `uxresearchJson` based on UX feedback from others
* removes the captcha during createCircle
* adds a new `contact` column to circles
* updates some of the helper interfaces to match the zeus ones. I imagine we might end up deleting these but its quite a challenge right now. 

Of note:
The old front end had some differences with create circle that are currently not carried forward, and warrant discussion:
* ~~the `uxresearchJson` that is part of createCircle - I think this includes the support email? I'm not 100% sure yet. ~~ This is deleted and replaced w/ new contact field
* ~~the `captcha_token` and server side verification of such~~ this is deleted
* ~~the `address` field was passed in by client before~~ this is deleted and determined on server side

Test plan:
* Create a circle in the UI, use the contact field and make sure that shows up in the db

### CODE REVIEW

**Comment by zashton** [comment]
> Fixes: #629
> 
> This also:
> 
> * Makes a number of columns that seemed non-nullable actually non-nullable to match the typescript interfaces in frontend
> * removes HASURA_ENABLED flag in FE since we want to default to using it and delete the old code
> 
> Of note: The old front end had some differences with create circle that are currently not carried forward, and warrant discussion:
> 
> * the `uxresearchJson` that is part of createCircle - I think this includes the support email? I'm not 100% sure yet.
> * the `captcha_token` and server side verification of such
> * the `address` field was passed in by client before
> 
> Test plan:
> 
> * Create a circle in the UI

1. The uxresearch field was requested to be removed based on 

https://www.notion.so/coordinape/Onboarding-Usability-Testing-Results-693be58b07454565aa4640966094edef#32b8261d52c4480bab239097d261daa1

2. captcha token had to be removed so circles can be created via API without going through a frontend.

3. Previously a user could create a circle without having a user record in the DB, hence it was passing the address (validated with signature) . Our current backend requires the user to have a profile record before creating a circle so the address is not necessary now.

**Review by levity — APPROVED** [review]

File: src/lib/gql/mutations.ts [hunk 1]
```diff
@@ -93,3 +95,60 @@ export const deleteCircleIntegration = async (id: number) =>
   client.mutate({
     delete_circle_integrations_by_pk: [{ id }, { id: true }],
   });
+
+export const createCircle = async (
+  address: string,
+  params: CreateCircleParam,
+  // TODO: these two args are no longer used in the hasura backend mutation, should we remedy this?
```
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply src/lib/gql/mutations.ts]: its removed now but changed to a contact filed

**Review by zashton — CHANGES_REQUESTED** [review]
main issue is when protocol is selected during create circle validation fails, otherwise its all good.

File: src/pages/CreateCirclePage/CreateCirclePage.tsx [hunk 2]
```diff
@@ -138,24 +133,9 @@ export const SummonCirclePage = () => {
...
-              { ...params },
-              captcha_token,
-              JSON.stringify({
-                address: myAddress,
-                research_org_link,
-                research_contact,
-                ...params,
-              })
-            );
+            const newCircle = await createCircle({ ...params });
```
[the remark this answers is not in the document]
Comment (topocount) [reply src/pages/CreateCirclePage/CreateCirclePage.tsx:138]: @zashton I think this is an issue with the zod validation, where it isn't logically identical to how laravel works right now: 
https://github.com/coordinape/coordinape-backend/blob/09c921ba25679451132a5c30852f4a522fd7cbc9/app/Http/Requests/CreateCircleRequest.php#L73-L74

As I understand it, either is required if the other isn't present, but this isn't and XOR gate where if one is present the other cannot be

[the remark this answers is not in the document]
Comment (zashton) [reply src/pages/CreateCirclePage/CreateCirclePage.tsx:138]: Yeah my bad, told @CryptoGraffe its a good idea to remove that part of the zod validation and ignore the `name` field if its sent together with the `protocol_id`

File: src/lib/gql/mutations.ts [hunk 3]
```diff
@@ -105,3 +107,54 @@ export const logout = async (): Promise<boolean> => {
...
+export const createCircle = async (
+  params: CreateCircleParam
+): Promise<IApiCircle> => {
+  const { createCircle } = await client.mutate({
+    createCircle: [
+      {
+        payload: params,
+      },
+      {
+        id: true,
```
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply src/lib/gql/mutations.ts:120]: ok

File: src/pages/CreateCirclePage/CreateCirclePage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply src/pages/CreateCirclePage/CreateCirclePage.tsx:138]: Ok i fixed this enough to make this work. Topo has a more thorough PR.

File: hasura/migrations/default/1647493316251_alter_table_public_circles_alter_column_created_at/down.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+alter table "public"."circles" alter column "created_at" drop not null;
```
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply hasura/migrations/default/1647493316251_alter_table_public_circles_alter_column_created_at/down.sql]: ok i did this, thanks for the tip

**Comment by CryptoGraffe** [comment]
Ok @topocount and @zashton I fixed the validation error, which led to me cleaning up a bunch of error handling so its all 100% centralized. I was unhappy with the zod error making it back to the UI.

**Review by zashton — APPROVED** [review]
validation issue is fixed lgtm

**Pull request merged**